apps/judicial-system/web/src/routes/Court/Indictments/Conclusion/Conclusion.tsx

```diff
@@ -80,76 +81,202 @@ const Conclusion: React.FC = () => {
     workingCase.id,
   )
 
-  const handleRedistribution = useCallback(async () => {
-    const transitionSuccessful = await transitionCase(
-      workingCase.id,
-      CaseTransition.REDISTRIBUTE,
-    )
+  const handleRedistribution = useCallback(
+    async (destination: string) => {
+      const transitionSuccessful = await transitionCase(
+        workingCase.id,
+        CaseTransition.REDISTRIBUTE,
+      )
 
-    if (transitionSuccessful) {
-      router.push(constants.CASES_ROUTE)
-    } else {
-      toast.error(formatMessage(errors.transitionCase))
-    }
-  }, [transitionCase, workingCase.id, formatMessage])
-
-  const handleCompletion = useCallback(async () => {
-    await setAndSendCaseToServer(
-      [
-        {
-          indictmentRulingDecision: selectedDecision,
-          force: true,
-        },
-      ],
+      const success = await setAndSendCaseToServer(
+        [
+          {
+            indictmentDecision: IndictmentDecision.REDISTRIBUTING,
+            force: true,
+          },
+        ],
+        workingCase,
+        setWorkingCase,
+      )
+
+      if (!transitionSuccessful || !success) {
+        return
```

Comment:
Bingo, that's correct, I think 😅 Did you mean to say 

"Perform the update first and only perform the transition if the update succeeds"? 

We want to perform both ideally, right?

Reply:
Yes, that's right.

1. Perform and wait for the update.
2. If the update failed, do nothing else, othewise perform and wait for the transition.
3. If the transition fails, do nothing else, otherwise navigate to the desired page.

Each failure should still show a toaster.